modules/swagger-codegen/src/main/resources/php/api.mustache

```diff
@@ -122,22 +124,38 @@ class {{classname}} {
         // for HTTP post (form)
         $httpBody = $formParams;
       }
-
-      // authentication setting, if any
-      $authSettings = array({{#authMethods}}'{{name}}'{{#hasMore}}, {{/hasMore}}{{/authMethods}});
-
+      {{#authMethods}}{{#isApiKey}}
```

Comment:
if this approach and the original approach both work, I would recommend keeping the authentication logic in ApiClient to avoid repeating the code in the endpoint method and to make it consistent with other API client (e.g, Python, C#, Objc, etc)